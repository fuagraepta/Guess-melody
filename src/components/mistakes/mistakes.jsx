import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {
  getMistakes,
  getErrorCount
} from '../../reducer/game/selectors.js';

const Mistakes = (props) => {
  const {
    mistakes,
    errorCount
  } = props;

  const correctAnswers = new Array(errorCount - mistakes).fill(`correct`);
  const wrongAnswers = new Array(mistakes).fill(`wrong`);

  return (
    <div className="game__mistakes">
      {correctAnswers.map((answer, i) => {
        return (
          <div key = {`${answer}- ${i}`} className={answer}></div>
        );
      })}
      {wrongAnswers.map((answer, i) => {
        return (
          <div key = {`${answer}- ${i}`} className={answer}></div>
        );
      })}
    </div>
  );
};


Mistakes.propTypes = {
  mistakes: PropTypes.number.isRequired,
  errorCount: PropTypes.number.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  mistakes: getMistakes(state),
  errorCount: getErrorCount(state)
});

export {Mistakes};
export default connect(mapStateToProps)(Mistakes);
