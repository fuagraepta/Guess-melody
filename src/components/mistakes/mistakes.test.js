import React from 'react';
import renderer from 'react-test-renderer';

import {Mistakes} from './mistakes.jsx';

describe(`The element is rendered correctly`, () => {
  it(`Mistakes element correctly renders if all answers correct`, () => {
    const mistakesElement = renderer
    .create(<Mistakes
      mistakes = {0}
      errorCount = {3}
    />);

    expect(mistakesElement).toMatchSnapshot();
  });

  it(`Mistakes element correctly renders if a few answers is incorrect`, () => {
    const mistakesElement = renderer
    .create(<Mistakes
      mistakes = {2}
      errorCount = {3}
    />);

    expect(mistakesElement).toMatchSnapshot();
  });
});
