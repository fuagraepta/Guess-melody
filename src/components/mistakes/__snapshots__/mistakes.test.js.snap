// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The element is rendered correctly Mistakes element correctly renders if a few answers is incorrect 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="correct"
  />
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;

exports[`The element is rendered correctly Mistakes element correctly renders if all answers correct 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="correct"
  />
  <div
    className="correct"
  />
  <div
    className="correct"
  />
</div>
`;
